source/administration/legacy-upgrade.md

```diff
@@ -1,17 +1,109 @@
+5. Restore the state of your server
+      1. Copy the backed up version of `config.json` in place of the default `config.json`.
+6. Start your server and address any setting changes relevant in the latest version of Mattermost
+      1. Run `sudo start mattermost`.
+      2. Opening the **System Console** and saving a change will upgrade your `config.json` schema to the latest version using default values for any new settings added.
+7. If you have TLS set up on your Mattermost server, you must activate the CAP_NET_BIND_SERVICE capability to allow the new Mattermost binary to bind to low ports.
+
+      1. `cd {install-path}`
+      2. `sudo setcap CAP_NET_BIND_SERVICE=+ep ./bin/platform`
+
+After the server is upgraded, users might need to refresh their browsers to experience any new features.
+
+## Upgrade Enterprise Edition from 3.0.0 and later
+
+### Important Notices
+
+1. Security related changes were made in 3.9.0 that cause any previously created team invite links, password reset links, and email verification links to no longer work. You must update any place where you have published these links.
+
+2. Security-related changes were made in 3.8.0 that require you to verify settings in the System Console before upgrading from version 3.7.0 and earlier to any version greater than 3.8.0
+
+  1. In the GENERAL section of the System Console, click **Configuration** and make sure that the **Site URL** is specified. It must not be empty. For more information about SiteURL, see `Configuration Settings <config-settings.html#site-url>`_
+  2. In the GENERAL section of the System Console, click **Logging** and make sure that the **File Log Directory** field is either empty or has a directory path only. It must not have a filename as part of the path.
+
+3. Changes were made in 3.8.0 that require a change in the proxy configuration. If you're using NGINX:
+  1. Open the NGINX configuration file as root. The file is usually ``/etc/nginx/sites-available/mattermost`` but might be different on your system.
+  2. Locate the following line:
+     `location /api/v3/users/websocket {`
+  3. Replace the  line with ``location ~ /api/v[0-9]+/(users/)?websocket$ {``.
+
+  If you are using a proxy other than NGINX, make the equivalent change to that proxy's configuration.
+
+**To upgrade your system from 3.0.0 and later**:
+
+1. Download the latest version of Enterprise Edition server and note any compatibility procedures:
+      - Note: If there are config settings set for `"RestrictPublicChannelManagement"` and `"RestrictPrivateChannelManagement"`, they will be used as the default values for `"RestrictPublicChannelCreation"`, `"RestrictPrivateChannelCreation"`, `"RestrictPublicChannelDeletion"`, and `"RestrictPrivateChannelDeletion"` after upgrade.
```

Comment:
worth mentioning what versions are affected? (I think upgrading from 3.5 and earlier to 3.6 and later)